compiler/typecheck/TcErrors.hs

```diff
@@ -553,6 +556,9 @@ reportWanteds ctxt tc_lvl (WC { wc_simple = simples, wc_impl = implics })
     env = cec_tidy ctxt
     tidy_cts = bagToList (mapBag (tidyCt env) simples)
 
+    report0 = [ ("Daml error", is_daml_error,    False, mkGroupReporter mkDamlErr) ]
```

Comment:
IIUC `False` here means that subsequent errors will be reported, which would include the default GHC error "Couldn't match X with Y arising from a functional dependency etc", but I don't think we want that. I'm probably missing something though.

Reply:
hm that sounds interesting, please let me know your findings :)

Reply:
Figured it out - I was accidentally letting the incoming ctxt to reportWanteds suppress daml errors and subsequently "utterly wrong" errors, which meant that in some cases no error would be thrown, and the codegen would then panic because we had completely invalid code generation.

Now I've made sure that daml and utterly wrong errors cannot be suppressed by incoming ctxt, and that only daml errors can suppress utterly wrong errors.